aiida/orm/utils/links.py

```diff
@@ -122,37 +122,59 @@ def validate_link(source, target, link_type, link_label):
     if not isinstance(source, type_source) or not isinstance(target, type_target):
         raise ValueError('cannot add a {} link from {} to {}'.format(link_type, type(source), type(target)))
 
-    # Validate the outdegree of the source node
-    outgoing = source.get_outgoing(link_type=link_type)
+    def link_triple_exists(source, target, link_type, link_label):
+        """Return whether a link with the given type and label exists between the given source and target node.
+
+        :param source: node from which the link is outgoing
+        :param target: node to which the link is incoming
+        :param link_type: the link type
+        :param link_label: the link label
+        :return: boolean, True if the link triple exists, False otherwise
+        """
+        from aiida.orm import QueryBuilder
+
+        if target._incoming_cache and LinkTriple(source, link_type, link_label) in target._incoming_cache:  # pylint: disable=protected-access
+            return True
+
+        builder = QueryBuilder()
+        builder.append(Node, filters={'uuid': source.uuid}, project=['id'])
+        builder.append(Node, filters={'uuid': target.uuid}, edge_filters={'type': link_type.value, 'label': link_label})
+        builder.limit(1)
+
+        return builder.count() != 0
+
+    if outdegree == 'unique_triple' or indegree == 'unique_triple':
```

Comment:
Let's set a default value (e.g. to False) to avoid `NameError`s? Or better to leave it like this so we notice if we change the code and forget to define the variable?

Reply:
I'd rather raise indeed then the conditional below accidentally passing